### ISSUE TITLE
AssertionFailedError interrupts test tree in Console and Gradle

### ISSUE THREAD

**Issue by smoyer64** [junit-team/junit-framework#658]
## Overview

- (X) **Bug report.** When assertions fail during a test run and provide a multi-line message (as is common with the Hamcrest assertions and matchers, the test tree is interrupted when running via the console or Gradle.  This problem is also seen when ```assertAll()``` throws a ```MultipleFailuresError```.  The example below demonstrates both these failure conditions:

```
@Test
public void assertAllExample() {
	assertAll(
		() -> { throw new AssertionError(new Throwable("first thing went wrong")); },
		() -> { throw new AssertionError(new Throwable("second thing went wrong")); },
		() -> assertThat("ERROR! - something is broke", not(containsString("ERROR!")))
	);
}

@Test
public void assertOneExample() {
	assertTrue(false, "Wasn't expecting that!\n\nNow I've broken the tree");
}
```

The output from when these tests are run via Gradle is as follows:

```
│  ├─ example.testrule.ErrorCollectorJupiterDemo
│  │  ├─ assertAllExample ✘ Multiple Failures (3 failures)
        java.lang.Throwable: first thing went wrong
        java.lang.Throwable: second thing went wrong
        
Expected: not a string containing "ERROR!"
     but: was "ERROR! - something is broke"
│  │  ├─ assertOneExample ✘ Wasn't expecting that!

Now I've broken the tree
```

## Deliverables

- [ ] Continue the tree structure across line breaks similar to the following:

```
│  ├─ example.testrule.ErrorCollectorJupiterDemo
│  │  ├─ assertAllExample ✘ Multiple Failures (3 failures)
│  │  │    java.lang.Throwable: first thing went wrong
│  │  │    java.lang.Throwable: second thing went wrong
│  │  │    
│  │  │    Expected: not a string containing "ERROR!"
│  │  │    but: was "ERROR! - something is broke"
│  │  ├─ assertOneExample ✘ Wasn't expecting that!
│  │  │    
│  │  │    Now I've broken the tree
```

**Comment by marcphilipp** [comment]
Thanks for opening the issue! I'm putting it _up-for-grabs_ for now. Do you want to give it a go?

**Comment by smoyer64** [comment]
@marcphilipp ... I'm working on #343 at the moment (could you review what I've proposed) and had also started on #144.  Since #343 is still slated for the M4 release I'd like to finish that work before I commit to any further fixes.  After that I'd be happy to pick off some of the up-for-grabs (I may need to add the ```fail()``` methods in #656 to finish #343 though).

**Comment by sormuras** [comment]
As I'm still mentally inside the tree-rendering topic, I assigned this issue to myself.

**Comment by sormuras** [comment]
The [VerboseTreePrintingListener](https://github.com/junit-team/junit5/blob/master/junit-platform-console/src/main/java/org/junit/platform/console/tasks/VerboseTreePrintingListener.java#L128) already handles the multiline detail message. Will move some of that logic up the hierachy.

**Comment by sormuras** [comment]
@smoyer64 This should be fixed with #667 - can you please verify? Thanks in advance.

**Comment by smoyer64** [comment]
**CONFIRMED**

I commented out line 30 of the ErrorCollectorJupiterDemo (```@extensions.ExpectToFail```) and now see the following output when the ```assertAll()``` method is executed:

```
│  ├─ ErrorCollectorJupiterDemo
│  │  ├─ example() ✘ Multiple Failures (3 failures)
│  │  │            java.lang.Throwable: first thing went wrong
│  │  │            java.lang.Throwable: second thing went wrong
│  │  │  
│  │  │      Expected: not a string containing "ERROR!"
│  │  │           but: was "ERROR! - something is broke"
```
